### PR TITLE
jdsol spec cleanup

### PR DESCRIPTION
**Pull request by markt-asf**
Not complete yet. Text has been imported, the license updated and a first pass made to fix most of the formatting issues.

I plan to add at least two more commits: one to fix any remaining formatting issues and one to fix all the links that are currently broken. There will probably be a third commit when I do the final proof read.

### CODE REVIEW

**Comment by joakime** [comment]
Has the work been completed to your satisfaction? (the opening comment says it hasn't, and you haven't updated this with a comment saying you are ready for review)

**Comment by markt-asf** [comment]
Just the final proof read to do then the PR will be ready.

**Comment by markt-asf** [comment]
OK. Ready for review now.

File: jdsol-spec/src/main/asciidoc/jdsol.adoc [hunk 1]
```diff
@@ -1,6 +1,1240 @@
...
+=== General Format
+
+The SMAP consists of lines of
+http://www.unicode.org/unicode/standard/standard.html[Unicode] text, with a
+concrete representation of http://ietf.org/rfc/rfc2279.txt[UTF-8]. Line
+termination is with line-feed, carriage-return or carriage-return followed by
+line-feed. Because SMAPs are included in class files, size of the SMAP was an
+important constraint on the format chosen for them.
+
+=== Header
```
Comment (joakime) [inline jdsol-spec/src/main/asciidoc/jdsol.adoc]: Not sure how to correct this, but `header` is a reserved word in asciidoc.

This is easiest to see in the html generated page and the odd css behavior due to this reserved word, it causes alternate formatting for this specific section heading.

It renders as ...

``` html
<h3 id="header"><a class="anchor" href="#header"></a>5.2. Header</h3>
```

The `id="header"` portion is what messes with formatting.

File: jdsol-spec/src/main/asciidoc/jdsol.adoc [hunk 2]
```diff
@@ -1,6 +1,1240 @@
...
+
+[cols="10%,90%"]
+|===
+|JVMDI |https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jvmdi-spec.html#GetSourceDebugExtension[`GetSourceDebugExtension(jclass clazz, char **sourceDebugExtensionPtr)`]
+|JDWP |https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdwp/jdwp-protocol.html#JDWP_ReferenceType_SourceDebugExtension[`SourceDebugExtension`] Command (12) in the `ReferenceType` (2) Command Set
+|JDI |https://docs.oracle.com/javase/1.5.0/docs/guide/jpda/jdi/com/sun/jdi/ReferenceType.html#sourceDebugExtension()[`String sourceDebugExtension()`] in the `ReferenceType` interface
+|===
+
+[#attribute]
+=== `SourceDebugExtension` Class File Attribute
```
Comment (joakime) [inline jdsol-spec/src/main/asciidoc/jdsol.adoc]: This is a heading, drop the source/code/pre formatting on `SourceDebugExtension` (you don't do this elsewhere on headings)

File: jdsol-spec/src/main/asciidoc/jdsol.adoc (hunk 1 above)
Comment (markt-asf) [reply jdsol-spec/src/main/asciidoc/jdsol.adoc]: Fixed with a custom ID

File: jdsol-spec/src/main/asciidoc/jdsol.adoc (hunk 2 above)
Comment (markt-asf) [reply jdsol-spec/src/main/asciidoc/jdsol.adoc]: Done. It was only there because it came across in the conversion from the original.

**Pull request merged**